src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -124,7 +124,20 @@ protected boolean isLegacy(String legacy, String navigationMode) {
 				|| NavigationMode.PP.toString().equalsIgnoreCase(navigationMode);
 	}
 
+	/**
+	 * Fetches a COM ID for non-null featureSource and featureID.
+	 * If either parameter is null, an IllegalArgumentException is thrown.
+	 *
+	 * @param featureSource
+	 * @param featureID
+	 * @return May return null if the COM ID is not found.
+	 */
 	protected String getComid(String featureSource, String featureID) {
+
+		if (null == featureSource || null == featureID) {
+			throw new IllegalArgumentException("A featureSource and featureID are required");
+		}
+
```

Comment:
To distinguish null parameters (a server error) from a NOT FOUND comId, we need to explicitly test for null parameters.  In normal use, the incoming parameters are never null, so if either are null it is really a server error,

Reply:
As a style note, all four places that this method are called force a value for both parameters. This could be viewed as unnecessary duplicate code.

Reply:
Agreed, a null cannot come through from the end points.  But this makes for a clear distinction between invalid arguments and Not Found, as well as makes it easier to detect it in the calling methods.